### PR TITLE
chore: Add nvmrc file

### PR DESCRIPTION
**Pull request by florian-richter**
## Context

- Adds a `.nvmrc` to automatically set the nvmrc to the latest stable node version (`node` tag which is currently v12) which avoids potential problems with `@sap/xsenv@3.0.0` which expects version "^8.0.0 || ^10.0.0 || ^12.0.0" 
- Replaces tabs in `.madgerc` with 2 spaces
- Fixes a typo in the PR template

## Definition of Done

Please consider all items and remove only if not applicable.

- N/A - Tests created/adjusted for your changes.
- N/A Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [x] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- N/A If applicable: Properly documented (JSDoc of public API)
- [x] If applicable: Check if ~~`node run doc`~~ `yarn run doc` still works.

### CODE REVIEW

**Comment by florian-richter** [comment]
**Question:** Should the release notes be updated for a change like this?

File: .madgerc [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
-    "fileExtensions" : ["ts"],
```
Comment (marikaner) [inline .madgerc:2]: Why?

**Review by marikaner — COMMENTED** [review]
Looks good, but why should we indent json like this?

**Comment by marikaner** [comment]
> **Question:** Should the release notes be updated for a change like this?

No, only consumer relevant changes.

File: .madgerc (hunk 1 above)
Comment (florian-richter) [reply .madgerc:2]: `.madgerc` was using a mix of 4 spaces and tabs which rendered badly in my editor. Other jsons in the project are also using 2 spaces so I thought it is appropriate.

Comment (marikaner) [reply .madgerc:2]: Ah, I saw it wrong, I thought you changed it to 4 spaces.

**Review by marikaner — APPROVED** [review]
LGTM

**Pull request merged**